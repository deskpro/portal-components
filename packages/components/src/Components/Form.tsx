import * as React from 'react';
import { Form as FormikForm } from 'formik';
import classNames from 'classnames';

interface IProps {
  className?: string;
  showHover?: boolean;
  children:   React.ReactNode;
}

const Form = ({
  className = '',
  showHover = true,
  ...props
}: IProps) => (
  <FormikForm
    className={classNames('dp-pc_form', className, { 'show-hover': showHover })}
    {...props}
  />);



export default Form;
