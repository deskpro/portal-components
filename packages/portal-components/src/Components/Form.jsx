import React from 'react';
import { Form as FormikForm } from 'formik';

const Form = ({
  ...props
}) => <FormikForm {...props} />;

export default Form;
