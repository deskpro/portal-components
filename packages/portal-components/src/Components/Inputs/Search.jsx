import React from 'react';

import Input from './Input';

class Search extends Input
{
  constructor(props) {
    super(props);
    this.type = 'search';
  }

}

export default Search;