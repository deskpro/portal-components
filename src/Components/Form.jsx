import React from 'react';
import { Form as FormikForm } from 'formik';

const Form = ({
  ...props
}) => {
  return <FormikForm {...props} />
};

export default Form;