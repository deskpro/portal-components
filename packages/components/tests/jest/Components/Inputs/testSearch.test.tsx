import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { withFormik } from 'formik';
import Form from '../../../../src/Components/Form';
import Search from '../../../../src/Components/Inputs/Search';

describe('>>> Search --- Snapshot', () => {
  it('+++capturing Snapshot of Search', () => {
    const App = () => (
      <Form>
        <Search name="search" />
      </Form>
    );
    const FormikApp = withFormik({
      mapPropsToValues({
        search
      }) {
        return {
          search: search || 'Test search'
        };
      },
      handleSubmit(values, { setSubmitting }) {
        setSubmitting(false);
      }
    })(App);
    const renderedValue = renderer.create(<FormikApp />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
