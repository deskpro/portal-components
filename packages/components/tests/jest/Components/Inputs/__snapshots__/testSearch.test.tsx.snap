// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Search --- Snapshot +++capturing Snapshot of Search 1`] = `
<form
  className="dp-pc_form show-hover"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="dp-pc_field"
  >
    <label
      className="dp-pc_label"
      htmlFor="field_1"
    >
      
    </label>
    <input
      className="dp-pc_input"
      id="field_1"
      name="search"
      onBlur={[Function]}
      onChange={[Function]}
      required={false}
      type="search"
      value="Test search"
    />
  </div>
</form>
`;
